src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -7388,7 +7400,11 @@ and TcRecdExpr cenv overallTy env tpenv (inherits, withExprOpt, synRecdFields, m
         match flds with
         | [] -> []
         | _ ->
-            match BuildFieldMap cenv env hasOrigExpr overallTy flds mWholeExpr with
+            let tyIdent =
+                match withExprOpt with
+                | Some (SynExpr.Ident ident, _) -> Some ident
```

Comment:
Why does it check a specific shape of the tree and only works for a simple identifier?

It's possible to get an anonymous property from a property (`x.Prop`), a function call (`f ()`), an indexer (`x.[0]`, or `x[0]`), and so on. There are many more different shapes of a tree that we may get here, and we should not assume some specific one, nor should we hardcode a few popular ones, while disabling recovery and analysis for all other cases.

Reply:
You are right. I was just interested in the `idText`. I will have a look a better way to do this

Reply:
Updated as discussed